rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/Rename.rsc

```diff
@@ -625,21 +643,44 @@ set[Define] getCursorDefinitions(list[Tree] cursor, Tree(loc) getTree, TModel(Tr
     return {};
 }
 
-tuple[set[loc], set[loc]] findOccurrenceFiles(set[Define] defs, list[Tree] cursor, Tree(loc) getTree, Renamer r) {
-    set[loc] getSourceFiles() {
+private set[loc]() getSourceFiles(Renamer r) {
+    return set[loc]() {
         return {*find(srcFolder, "rsc")
             | wsFolder <- r.getConfig().workspaceFolders
             , srcFolder <- r.getConfig().getPathConfig(wsFolder).srcs
         };
-    }
+    };
+}
 
-    if ({IdRole role} := defs.idRole
-      && role notin {variableId(), patternVariableId(), moduleId()}) {
+tuple[set[loc], set[loc], set[loc]] findOccurrenceFiles(set[Define] defs, list[Tree] cursor, str newName, Tree(loc) getTree, Renamer r) {
+    if ({IdRole role} := defs.idRole) {
         <t, _> = asType(role);
-        return findOccurrenceFilesSymmetric(t, "<cursor[0]>", getSourceFiles, getTree);
+        name = "<cursor[0]>";
+        try {
+            // TODO Check if specific subtype of Tree is correct here
+            newNameFiles = findSortOccurrenceFiles(t, newName, getSourceFiles(r), getTree);
```

Comment:
shouldn't we make sure that we only look for `newNameFiles` based on the `role`?

Reply:
Indeed. I changed this now.